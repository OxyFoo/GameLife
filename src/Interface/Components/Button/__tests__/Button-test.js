import 'react-native';
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// Graphical components
import { Button } from '../index';

describe('[Component] Button', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Button>Test</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('event onPress', () => {
        const onPress = jest.fn();
        const { getByTestId } = render(
            <Button testID='button' onPress={onPress}>
                Test
            </Button>
        );

        const button = getByTestId('button');

        // Simulez les événements tactiles
        fireEvent(button, 'touchStart', {
            nativeEvent: { pageX: 100, pageY: 100, locationX: 50, locationY: 50 }
        });

        fireEvent(button, 'touchEnd', {
            nativeEvent: { pageX: 100, pageY: 100 }
        });

        // Vérifiez si onPress a été appelé
        expect(onPress).toHaveBeenCalledTimes(1);
    });

    it('event onPress disabled', () => {
        const onPress = jest.fn();
        const { getByTestId } = render(
            <Button testID='button' onPress={onPress} enabled={false}>
                Test
            </Button>
        );

        const button = getByTestId('button');
        fireEvent.press(button);

        expect(button).toBeTruthy();
        expect(onPress).toHaveBeenCalledTimes(0);
    });
});
