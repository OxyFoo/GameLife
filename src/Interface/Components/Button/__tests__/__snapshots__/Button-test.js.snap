// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Button renders correctly 1`] = `
<View
  onLayout={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderRadius": 8,
        "overflow": "hidden",
        "paddingHorizontal": 24,
        "paddingVertical": 16,
        "width": "100%",
      },
      {},
      null,
    ]
  }
>
  <BVLinearGradient
    angle={267}
    colors={
      [
        4287428607,
        4292583935,
      ]
    }
    endPoint={
      {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    useAngle={true}
  />
  <View
    pointerEvents="none"
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
        },
        {
          "justifyContent": "center",
          "opacity": 1,
        },
        {},
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Hind Vadodara",
            "margin": 0,
            "padding": 0,
            "textAlign": "center",
          },
          {
            "color": "#384065",
            "fontSize": 16,
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      Test
    </Text>
  </View>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      {
        "aspectRatio": 1,
        "backgroundColor": "#000000",
        "borderRadius": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "top": 0,
        "transform": [
          {
            "translateX": 0,
          },
          {
            "translateY": 0,
          },
          {
            "scale": 0,
          },
        ],
        "width": 2,
      }
    }
  />
</View>
`;
